griptape/drivers/image_generation/openai_dalle_image_generation_driver.py

```diff
@@ -0,0 +1,74 @@
+import base64
+import openai
+from typing import Optional, Literal, Union, get_args
+from attr import field, Factory, define
+from griptape.artifacts import ImageArtifact
+from griptape.drivers import BaseImageGenerationDriver
+
+
+@define
+class OpenAiDalleImageGenerationDriver(BaseImageGenerationDriver):
+    """Driver for OpenAI DALLE image generation API.
+
+    Attributes:
+        model: OpenAI DALLE model, for example 'dall-e-2' or 'dall-e-3'.
+        api_type: OpenAI API type, for example 'open_ai' or 'azure'.
+        api_version: API version.
+        base_url: API URL.
+        api_key: OpenAI API key.
+        organization: OpenAI organization ID.
+        style: Optional and only supported for dall-e-3, can be either 'vivid' or 'natural'.
+        quality: Optional and only supported for dall-e-3. Accepts 'standard', 'hd'.
+        image_size: Size of the generated image. Must be one of the following, depending on the requested model:
+            dall-e-2: [256x256, 512x512, 1024x1024]
+            dall-e-3: [1024x1024, 1024x1792, 1792x1024]
+        response_format: The response format. Currently only supports 'b64_json' which will return
+            a base64 encoded image in a JSON object.
+    """
+
+    api_type: str = field(default=openai.api_type, kw_only=True)
+    api_version: Optional[str] = field(default=openai.api_version, kw_only=True)
+    base_url: str = field(default=None, kw_only=True)
+    api_key: Optional[str] = field(default=None, kw_only=True)
+    organization: Optional[str] = field(default=openai.organization, kw_only=True)
+    client: openai.OpenAI = field(
+        default=Factory(
+            lambda self: openai.OpenAI(api_key=self.api_key, base_url=self.base_url, organization=self.organization),
+            takes_self=True,
+        )
+    )
+    style: Optional[str] = field(default=None, kw_only=True)
+    quality: Union[Literal["standard"], Literal["hd"]] = field(default=Literal["standard"], kw_only=True)
+    image_size: Union[
+        Literal["256x256"], Literal["512x512"], Literal["1024x1024"], Literal["1024x1792"], Literal["1792x1024"]
+    ] = field(default=Literal["512x512"], kw_only=True)
```

Comment:
I think it's somewhat unavoidable to expose SDK patterns in Drivers since their primary purpose is to sit right on top of the SDK/API. Even in the case of `image_size`, a seemingly universal field, it seems non-straightforward to implement it at the `BaseImageGenerationDriver` level. I think providing helpful type hints is a good enough solution.

Reply:
It's certainly less maintenance to do it this way. If we define and enforce a standard Griptape interface style, the intermediate layer would be a constant source of frustration as we'd have to re-define (or somehow handle) all possible options to some degree. On the other hand, we require the user do some research and have knowledge of the underlying dependency (OpenAI SDK, Leonardo API) before use. Not at all unreasonable, but a bit annoying.